core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -150,7 +150,19 @@ extends MemStore with StrictLogging {
     if (endOffset < startOffset) Observable.empty
     else {
       var targetOffset = startOffset + reportingInterval
-      stream.map(shard.ingest(_)).collect {
+      var startOffsetValidated = false
+      stream.map { r =>
+        if (!startOffsetValidated) {
+          if (r.offset > startOffset) {
+            val offsetsNotRecovered = r.offset - startOffset
+            logger.error(s"Could not recover dataset=$dataset shard=$shardNum from check pointed offset possibly " +
+                         s"because of retention issues. offsetsNotRecovered=$offsetsNotRecovered")
+            shard.shardStats.offsetsNotRecovered.increment(offsetsNotRecovered)
```

Comment:
This is fine, though I feel like in the future these should be "events" and not metrics.

Reply:
agreed.